compiler/codegen/OMRCodeGenerator.hpp

```diff
@@ -482,6 +482,12 @@ class OMR_EXTENSIBLE CodeGenerator
 
    TR::Recompilation *allocateRecompilationInfo() { return NULL; }
 
+   /*
+    * This determines if it is necessary to emit a prefetch instruction.
+    * If so, it also emits the prefetch instruction.
+    */
```

Comment:
Could you add the documentation for each of these parameters and what they are for so that a downstream implementing project knows what to do?  I must admit I'm confused by `targetRegister` because the one place this is used in OMR the argument passed in is named `source` !  Thanks.

Reply:
I added a comment saying what each parameter was. The names I used were the same as the ones in the original addPrefetch function that got moved to OpenJ9 and renamed to insertPrefetchIfNecessary. Your specific example of passing in "source" as the "targetRegister" was something that was already there. I think it is because it is a source register inside l2aEvaluator but it is used to generate the prefetch target inside insertPrefetchIfNecessary.

Reply:
Thanks.  But please use Doxygen syntax (i.e., `@brief`, `@param`, `@return` tags) in comments, and document the API in the header file (`OMRCodeGenerator.hpp`) not the implementation (`.cpp`) file.

Reply:
I moved the comment to the hpp file and changed it to use Doxygen syntax.